### DISCUSSION TITLE
DataGridTextColumn.width isn't implemented!

### DISCUSSION THREAD

**Discussion by Aytharn** [fsprojects/Avalonia.FuncUI#361] [Q&A]
ChatGTP 4.0 tells me:
You're correct, my apologies for the confusion. In Avalonia.FuncUI, the DataGridTextColumn.width property does not exist. Instead, you can use the DataGridTextColumn.Width property in Avalonia, but this property is not directly accessible in Avalonia.FuncUI
[4](https://docs.avaloniaui.net/docs/next/reference/controls/datagrid/datagridcolumns)
[5](https://github.com/AvaloniaUI/Avalonia/discussions/7612)
.
Currently, Avalonia.FuncUI does not support setting the width of DataGrid columns directly. This is a limitation of the Avalonia.FuncUI library, which provides a functional wrapper around Avalonia but does not expose all properties of the underlying controls
[5](https://github.com/AvaloniaUI/Avalonia/discussions/7612)
.
As a workaround, you can define your DataGrid in XAML where you have access to all properties, and then use Avalonia.FuncUI for the rest of your UI.

Is it correct?

**Comment by JaggerJo** [comment]
FuncUI is quite flexible. I've added bindings for the `width` property in 5 minutes: https://github.com/fsprojects/Avalonia.FuncUI/commit/a4b6ca6b592e08fdaad195a7be949edc1e06c7ff

Don't ask chat GPT, it has no clue. Always verify things yourself.

**Comment by Aytharn** [comment]
For instance I did: 

```
type DataGrid with
    static member headersVisibility<'t when 't :> DataGrid>(value: DataGridHeadersVisibility) : IAttr<'t> =
        AttrBuilder<'t>.CreateProperty<DataGridHeadersVisibility>(
            name = "headersVisibility",
            value = value,
            getter = ValueSome (fun grid -> grid.HeadersVisibility),
            setter = ValueSome (fun (grid, value) -> grid.HeadersVisibility <- value),
            comparer = ValueNone
        )
```
